service/core.go

```diff
@@ -193,3 +199,21 @@ func (c *Core) nextTaskID() int64 {
 func (c *Core) requestLogger(r *http.Request) *zap.Logger {
 	return c.logger.With(zap.String("request_id", api.RequestIDFromContext(r.Context())))
 }
+
+func (c *Core) publishEvent(event string, data string) {
+	eventPayload := fmt.Sprintf("event: %s\n", event)
+	dataLines := strings.Split(data, "\n")
```

Comment:
You don't have any multi-line data at this point, I would just keep this simple:
```
payload := fmt.Sprintf("event: %s\ndata: %s", event, data)
```